File: my_app/lib/models/customer.dart
class Customer {
  int customerNr;   // matches CustomerNr in Go
  String firm;
  String phoneNr;   // matches PhoneNr in Go
  String mail;
  String location;

  Customer({
    required this.customerNr,
    required this.firm,
    required this.phoneNr,
    required this.mail,
    required this.location,
  });

  factory Customer.fromJson(Map<String, dynamic> json) {
    return Customer(
      customerNr: json['customer_nr'] ?? 0,    // Map Go field CustomerNr to customerNr
      firm: json['firm'] ?? 'Unknown',
      phoneNr: json['phone_nr'] ?? 'Unknown',  // Map Go field PhoneNr to phoneNr
      mail: json['mail'] ?? 'Unknown',
      location: json['location'] ?? 'Unknown',
    );
  }
}

File: my_app/go/models/customer.go
package models

// Customer represents a customer record in the database
type Customer struct {
	CustomerNr int32  `json:"customer_nr"`
	Firm       string `json:"firm"`
	PhoneNr    string `json:"phone_nr"`
	Mail       string `json:"mail"`
	Location   string `json:"location"`
}
